app/main.py

```diff
@@ -1,3 +1,64 @@
+import math
+
+
 class Vector:
-    # write your code here
-    pass
+    def __init__(self, x, y):
+        self.x = round(x, 2)
+        self.y = round(y, 2)
+
+    def __add__(self, other):
+        return Vector(
+            self.x + other.x,
+            self.y + other.y
+        )
+
+    def __sub__(self, other):
+        return Vector(
+            self.x - other.x,
+            self.y - other.y
+        )
+
+    def __mul__(self, other):
+        if isinstance(other, (float, int)):
+            return Vector(
+                self.x * other,
+                self.y * other
+            )
+        else:
+            dot_product = (self.x * other.x) + (self.y * other.y)
+            return dot_product
+
+    @classmethod
+    def create_vector_by_two_points(cls, start_point: tuple, end_point: tuple):
+        start_point = Vector(start_point[0], start_point[1])
+        end_point = Vector(end_point[0], end_point[1])
+        return end_point.__sub__(start_point)
```

Comment:
not the best way of using magic methods; not for nothing they are magic :) fix this throughout your code